### PR TITLE
Raspberry Pi/arm64 compatible builds

### PR DESCRIPTION
**Pull request by shanemadden**
I'm really not sure if there's a more 'proper' way to organize the infrastructure for an alternate architecture in this situation, but, maybe it's a start?

Added alternate compose file using arm64v8 builds of redis/mongo, as well as alternate dockerfile using arm64v8 node base image.

Instructions for use (tested on my pi4 locally):

```sh
# install os, must be 64-bit! recommend a fresh install:
# https://downloads.raspberrypi.com/raspios_lite_arm64/images/raspios_lite_arm64-2023-12-11/2023-12-11-raspios-bookworm-arm64-lite.img.xz

# install git
apt-get install git

# install docker-ce (remove existing packages if already installed)
curl -fsSL https://get.docker.com -o get-docker.sh
sh get-docker.sh

cd /opt
# git clone https://github.com/Jomik/screeps-server.git
# WORKAROUND, in branch temporarily
git clone https://github.com/shanemadden/screeps-server.git
cd screeps-server

# WORKAROUND, in branch temporarily
git checkout pi

cp .env.sample .env
# edit .env, add steam API key

docker compose -f docker-compose-arm64.yml up --build -d
```

### CODE REVIEW

File: docker-compose-arm64.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+version: "3"
+services:
+  screeps:
+    platform: linux/arm64
```
[the remark this answers is not in the document]
Comment (shanemadden) [reply docker-compose-arm64.yml]: My research seemed to indicate it wasn't needed (though I don't own an apple silicon system so I can't confirm from that perspective, unfortunately) - gonna try dropping it and see if that works here!

[the remark this answers is not in the document]
Comment (Jomik) [reply docker-compose-arm64.yml]: I'll test it. I have one 👍

[the remark this answers is not in the document]
Comment (Jomik) [reply docker-compose-arm64.yml]: Builds and runs fine on your branch on my Mac M3.

**Review by Jomik — COMMENTED** [review]
Will you delete `docker-compose-arm64.yml` and add the pins to the repo `docker-compose.yml`? Then I think we can merge this. We can see if the pushed image works for your RPi then :)

**Review by Jomik — APPROVED** [review]

**Pull request merged**